chips/nrf52/src/uart.rs

```diff
@@ -270,32 +272,67 @@ impl Uarte {
                 return;
             }
 
-            self.remaining_bytes.set(rem - tx_bytes);
+            self.tx_remaining_bytes.set(rem - tx_bytes);
             self.offset.set(tx_bytes);
 
-            if self.remaining_bytes.get() == 0 {
+            if self.tx_remaining_bytes.get() == 0 {
                 // Signal client write done
                 self.client.get().map(|client| {
-                    self.buffer.take().map(|buffer| {
-                        client.transmit_complete(buffer, kernel::hil::uart::Error::CommandComplete);
+                    self.tx_buffer.take().map(|tx_buffer| {
+                        client.transmit_complete(
+                            tx_buffer,
+                            kernel::hil::uart::Error::CommandComplete,
+                        );
                     });
                 });
             }
             // Not all bytes have been transmitted then update offset and continue transmitting
             else {
-                self.set_dma_pointer_to_buffer();
+                self.set_tx_dma_pointer_to_buffer();
                 regs.task_starttx.write(Task::ENABLE::SET);
                 self.enable_tx_interrupts();
             }
         }
+
+        if self.rx_ready() {
+            self.disable_rx_interrupts();
+            regs.event_endrx.write(Event::READY::CLEAR);
+            //Get the number of bytes in the buffer
+            let rx_bytes = regs.rxd_amount.get() as usize;
+            let rem = self.rx_remaining_bytes.get();
+            //should check if rx_bytes > 0 (because we'll flush the FIFO and it will flag
+            //ENDRX even if it's empty)
+            if rx_bytes > 0 {
+                self.rx_remaining_bytes.set(rem - rx_bytes);
```

Comment:
`rem - rx_bytes` can actually overflow here if `rx_bytes` > `rem` so I would be more satisfied using:

```self.rx_remaining_bytes.set(rem.saturating_sub(rx_bytes));``` or check it as we do in tx-mode!

Reply:
done!